_dev/src/ts/api/RequestHandler.ts

```diff
@@ -5,30 +5,55 @@ import Hydration from '../utils/Hydration';
 export class RequestHandler {
   private currentRequestAbortController: AbortController | null = null;
 
-  public post(route: string, data = new FormData(), fromPopState?: boolean) {
+  /**
+   * @public
+   * @param {string} route - Target route for the POST request.
+   * @param {FormData}[data=new FormData()] - Form data to send with the request by default we send FormData with admin dir required by backend.
+   * @param {boolean} [fromPopState] - Indicates if the request originated from a popstate event need by hydration.
+   * @returns {Promise<void>}
+   * @description Sends a POST request to the specified route with optional data and pop state indicator. Cancels any ongoing request before initiating a new one.
+   */
+  public async post(
+    route: string,
+    data: FormData = new FormData(),
+    fromPopState?: boolean
+  ): Promise<void> {
+    // Cancel any previous request if it exists
     if (this.currentRequestAbortController) {
       this.currentRequestAbortController.abort();
     }
 
+    // Create a new AbortController for the current request (used to cancel previous request)
     this.currentRequestAbortController = new AbortController();
     const { signal } = this.currentRequestAbortController;
 
+    // Append admin dir required by backend
     data.append('dir', window.AutoUpgradeVariables.admin_dir);
 
-    baseApi
-      .post('', data, {
-        params: { route: route },
+    try {
+      const response = await baseApi.post('', data, {
+        params: { route },
         signal
-      })
-      .then((response) => {
-        const data = response.data as ApiResponse;
-        this.handleResponse(data, fromPopState);
       });
+
+      const responseData = response.data as ApiResponse;
+      await this.#handleResponse(responseData, fromPopState);
+    } catch (error) {
+      // TODO: catch errors
+      console.error(error);
+    }
```

Comment:
Is this `try` = `catch` really necessary? It silences the error and the callee is now unaware of what's happening.

Reply:
I did this so that I could implement the error page in case of 500